homeassistant/components/androidtv/media_player.py

```diff
@@ -192,68 +215,118 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
         else:
             device_name = "Android TV / Fire TV device"
 
-        _LOGGER.warning("Could not connect to %s at %s %s", device_name, host, adb_log)
+        _LOGGER.warning(
+            "Could not connect to %s at %s %s", device_name, address, adb_log
+        )
         raise PlatformNotReady
 
-    if host in hass.data[ANDROIDTV_DOMAIN]:
-        _LOGGER.warning("Platform already setup on %s, skipping", host)
+    if address in hass.data[ANDROIDTV_DOMAIN]:
+        _LOGGER.warning("Platform already setup on %s, skipping", address)
+        return
+
+    device_args = [
+        aftv,
+        config[CONF_NAME],
+        config[CONF_APPS],
+        config[CONF_GET_SOURCES],
+        config.get(CONF_TURN_ON_COMMAND),
+        config.get(CONF_TURN_OFF_COMMAND),
+    ]
+
+    if aftv.DEVICE_CLASS == DEVICE_ANDROIDTV:
+        device = AndroidTVDevice(*device_args)
+        device_name = config.get(CONF_NAME, "Android TV")
     else:
-        if aftv.DEVICE_CLASS == DEVICE_ANDROIDTV:
-            device = AndroidTVDevice(
-                aftv,
-                config[CONF_NAME],
-                config[CONF_APPS],
-                config[CONF_GET_SOURCES],
-                config.get(CONF_TURN_ON_COMMAND),
-                config.get(CONF_TURN_OFF_COMMAND),
-            )
-            device_name = config[CONF_NAME] if CONF_NAME in config else "Android TV"
-        else:
-            device = FireTVDevice(
-                aftv,
-                config[CONF_NAME],
-                config[CONF_APPS],
-                config[CONF_GET_SOURCES],
-                config.get(CONF_TURN_ON_COMMAND),
-                config.get(CONF_TURN_OFF_COMMAND),
-            )
-            device_name = config[CONF_NAME] if CONF_NAME in config else "Fire TV"
+        device = FireTVDevice(*device_args)
+        device_name = config.get(CONF_NAME, "Fire TV")
+
+    add_entities([device])
+    _LOGGER.debug("Setup %s at %s %s", device_name, address, adb_log)
+    hass.data[ANDROIDTV_DOMAIN][address] = device
+
+    if not hass.services.has_service(ANDROIDTV_DOMAIN, SERVICE_ADB_COMMAND):
```

Comment:
Since all services are registered on first invocation of setup_platform, can't we just `return` if it has the ADB command service ?

Reply:
Fixed in https://github.com/home-assistant/home-assistant/pull/30086/commits/f2aa31726067017bce8d61a6477ef80d31cbd3f4